acquire/dynamic/windows/handles.py

```diff
@@ -0,0 +1,259 @@
+
+kernel32 = ctypes.windll.kernel32
+kernel32.OpenProcess.restype = HANDLE
+kernel32.DuplicateHandle.argtypes = (HANDLE, HANDLE, HANDLE, ctypes.POINTER(HANDLE), DWORD, BOOL, DWORD)
+
+
+class DuplicateFilter(Filter):
+    def __init__(self) -> None:
+        super().__init__()
+        self.msgs = set()
+
+    def filter(self, record: LogRecord) -> bool:
+        msg = record.getMessage()
+        if show := msg not in self.msgs:
+            self.msgs.add(msg)
+
+        return show
+
+
+def get_handle_type_info(handle: SYSTEM_HANDLE_TABLE_ENTRY_INFO_EX) -> Optional[str]:
+    """Return type of handle.
+
+    Parameters:
+        handle: handle for which to return the type information.
+
+    Raises:
+        RuntimeError: Raised when the result of the object query is unknown (other than SUCCESS, LENGTH MISMATCH or
+        INVALID).
+    """
+    public_object_type_information = PUBLIC_OBJECT_TYPE_INFORMATION()
+    size = DWORD(ctypes.sizeof(public_object_type_information))
+    while True:
+        result = ntdll.NtQueryObject(
+            handle,
+            OBJECT_INFORMATION_CLASS.ObjectTypeInformation,
+            ctypes.byref(public_object_type_information),
+            size,
+            None,
+        )
+
+        if result == NtStatusCode.STATUS_SUCCESS:
+            return str(public_object_type_information.Name)
+        elif result == NtStatusCode.STATUS_INFO_LENGTH_MISMATCH:
+            size = DWORD(size.value * 4)
+            ctypes.resize(public_object_type_information, size.value)
+        elif result == NtStatusCode.STATUS_INVALID_HANDLE:
+            return None
+        else:
+            raise RuntimeError(hex(result))
+
+
+def open_process(pid: int) -> int:
+    """Obtain a handle for the given PID.
+
+    More info: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-openprocess
+
+    Parameters:
+        pid: integer that represents the process ID.
+
+    Raises:
+        OpenProcessError: Raies when the System Idle Process, the System Process or one of the CSRSS processes are tried
+        to be opened.
+    """
+    kernel32.SetLastError(0)
+
+    h_process = kernel32.OpenProcess(
+        ProcessAccess.PROCESS_DUP_HANDLE,
+        False,
+        pid,
+    )
+
+    error = kernel32.GetLastError()
+    if error in [ErrorCode.ERROR_INVALID_PARAMETER, ErrorCode.ERROR_ACCESS_DENIED]:
+        raise OpenProcessError(
+            f"Likely tried opening the System Idle Process, the System Process or one of the Client Server Run-Time"
+            f"Subsystem (CSRSS) processes [pid: {pid}]"
+        )
+
+    # No valid handle could be obtained, display the error code
+    if h_process == 0:
+        raise OpenProcessError(f"OpenProcess Error: 0x{error:x} [pid: {pid}]")
+
+    return h_process
+
+
+def _get_file_name_thread(h_file: HANDLE, q: Queue):
+    iob = IO_STATUS_BLOCK()
+    file_name_information = ctypes.create_string_buffer(0x1000)
```

Comment:
As GitHub hides the original conversation as it is outdated somehow, but this magic number could do with some comment about why it is 0x1000 and not another number wink